assets/javascript/apps/pipeline/nodes/widgets.tsx

```diff
@@ -1101,3 +1104,219 @@ function BuiltInToolsWidget(props: WidgetParams) {
     </InputField>
   );
 }
+
+export function TextEditorWidget(props: WidgetParams) {
+  const { parameterValues } = getCachedData();
+  const autocomplete_vars_list: string[] = Array.isArray(parameterValues.text_editor_autocomplete_vars)
+  ? parameterValues.text_editor_autocomplete_vars.map((v: Option) => v.value) : [];
+
+  const modalId = useId();
+  const [isDarkMode, setIsDarkMode] = useState(false);
+  const setNode = usePipelineStore((state) => state.setNode);
+
+  const onChangeCallback = (value: string) => {
+  setNode(
+    props.nodeId,
+    produce((draft) => {
+      draft.data.params[props.name] = value;
+    })
+  );
+};
+
+  const openModal = () =>
+    (document.getElementById(modalId) as HTMLDialogElement)?.showModal();
+
+  useEffect(() => {
+    const updateTheme = () =>
+      setIsDarkMode(document.body.getAttribute("data-theme") === "dark");
+    updateTheme();
+    const observer = new MutationObserver(updateTheme);
+    observer.observe(document.body, { attributes: true });
+    return () => observer.disconnect();
+  }, []);
+
+  const label = (
+    <>
+      {props.label}
+        <div className="tooltip tooltip-left" data-tip={`Expand ${props.label}`}>
+        <button
+          type="button"
+          className="btn btn-xs btn-ghost float-right"
+          onClick={openModal}
+        >
+          <i className="fa-solid fa-expand-alt"></i>
+        </button>
+      </div>
+    </>
+  );
+
+  return (
+    <>
+      <InputField
+        label={label}
+        help_text={props.helpText}
+        inputError={props.inputError}
+      >
+        <div className="relative w-full">
+          <textarea
+            className="textarea textarea-bordered resize-none textarea-sm w-full !bg-white !text-black"
```

Comment:
Textarea colors are controlled by the theme
```suggestion
            className="textarea textarea-bordered resize-none textarea-sm w-full"
```